npm install failed

**Angular version* 7 *:

CLI 8.3 (latest)

**ng-multiselect-dropdown version* latest*:

**Description of issue* i have clone repositery and run npm installl *:

**Steps to reproduce**:

**Expected result*it should build sucessfully*:

**Actual result**:

**Demo**: Please share sample code link using StackBlitz or codesandbox

Any relevant code:
```
npm WARN invalid config registry="http"
npm WARN invalid config Must be a full url with 'http://'
npm WARN invalid config registry="http"
npm WARN invalid config Must be a full url with 'http://'
npm WARN deprecated browserslist@2.11.3: Browserslist 2 could fail on reading Browserslist >3.0 config used in other tools.

> node-sass@4.5.3 install C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\node-sass
> node scripts/install.js

Downloading binary from https://npm.taobao.org/mirrors/node-sass/v4.5.3/win32-x64-72_binding.node
Cannot download "https://npm.taobao.org/mirrors/node-sass/v4.5.3/win32-x64-72_binding.node":

HTTP error 404 Not Found

Hint: If github.com is not accessible in your location
      try setting a proxy via HTTP_PROXY, e.g.

      export HTTP_PROXY=http://example.com:1234

or configure npm proxy via

      npm config set proxy http://example.com:8080

> node-sass@4.5.3 postinstall C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\node-sass
> node scripts/build.js

Building: C:\Program Files\nodejs\node.exe C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\node-gyp\bin\node-gyp.js rebuild --verbose --libsass_ext= --libsass_cflags= --libsass_ldflags= --libsass_library=
gyp info it worked if it ends with ok
gyp verb cli [
gyp verb cli   'C:\\Program Files\\nodejs\\node.exe',
gyp verb cli   'C:\\Users\\umar.safeer\\Desktop\\test\\ng-multiselect-dropdown-master\\node_modules\\node-gyp\\bin\\node-gyp.js',
gyp verb cli   'rebuild',
gyp verb cli   '--verbose',
gyp verb cli   '--libsass_ext=',
gyp verb cli   '--libsass_cflags=',
gyp verb cli   '--libsass_ldflags=',
gyp verb cli   '--libsass_library='
gyp verb cli ]
gyp info using node-gyp@3.8.0
gyp info using node@12.13.1 | win32 | x64
gyp verb command rebuild []
gyp verb command clean []
gyp verb clean removing "build" directory
gyp verb command configure []
gyp verb download using dist-url https://npm.taobao.org/dist
gyp verb check python checking for Python executable "python2" in the PATH
gyp verb `which` failed Error: not found: python2
gyp verb `which` failed     at getNotFoundError (C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\which\which.js:13:12)
gyp verb `which` failed     at F (C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\which\which.js:68:19)
gyp verb `which` failed     at E (C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\which\which.js:80:29)
gyp verb `which` failed     at C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\which\which.js:89:16
gyp verb `which` failed     at C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\isexe\index.js:42:5
gyp verb `which` failed     at C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\isexe\windows.js:36:5
gyp verb `which` failed     at FSReqCallback.oncomplete (fs.js:158:21)
gyp verb `which` failed  python2 Error: not found: python2
gyp verb `which` failed     at getNotFoundError (C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\which\which.js:13:12)
gyp verb `which` failed     at F (C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\which\which.js:68:19)
gyp verb `which` failed     at E (C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\which\which.js:80:29)
gyp verb `which` failed     at C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\which\which.js:89:16
gyp verb `which` failed     at C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\isexe\index.js:42:5
gyp verb `which` failed     at C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\isexe\windows.js:36:5
gyp verb `which` failed     at FSReqCallback.oncomplete (fs.js:158:21) {
gyp verb `which` failed   stack: 'Error: not found: python2\n' +
gyp verb `which` failed     '    at getNotFoundError (C:\\Users\\umar.safeer\\Desktop\\test\\ng-multiselect-dropdown-master\\node_modules\\which\\which.js:13:12)\n' +
gyp verb `which` failed     '    at F (C:\\Users\\umar.safeer\\Desktop\\test\\ng-multiselect-dropdown-master\\node_modules\\which\\which.js:68:19)\n' +
gyp verb `which` failed     '    at E (C:\\Users\\umar.safeer\\Desktop\\test\\ng-multiselect-dropdown-master\\node_modules\\which\\which.js:80:29)\n' +
gyp verb `which` failed     '    at C:\\Users\\umar.safeer\\Desktop\\test\\ng-multiselect-dropdown-master\\node_modules\\which\\which.js:89:16\n' +
gyp verb `which` failed     '    at C:\\Users\\umar.safeer\\Desktop\\test\\ng-multiselect-dropdown-master\\node_modules\\isexe\\index.js:42:5\n' +
gyp verb `which` failed     '    at C:\\Users\\umar.safeer\\Desktop\\test\\ng-multiselect-dropdown-master\\node_modules\\isexe\\windows.js:36:5\n' +
gyp verb `which` failed     '    at FSReqCallback.oncomplete (fs.js:158:21)',
gyp verb `which` failed   code: 'ENOENT'
gyp verb `which` failed }
gyp verb check python checking for Python executable "python" in the PATH
gyp verb `which` failed Error: not found: python
gyp verb `which` failed     at getNotFoundError (C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\which\which.js:13:12)
gyp verb `which` failed     at F (C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\which\which.js:68:19)
gyp verb `which` failed     at E (C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\which\which.js:80:29)
gyp verb `which` failed     at C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\which\which.js:89:16
gyp verb `which` failed     at C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\isexe\index.js:42:5
gyp verb `which` failed     at C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\isexe\windows.js:36:5
gyp verb `which` failed     at FSReqCallback.oncomplete (fs.js:158:21)
gyp verb `which` failed  python Error: not found: python
gyp verb `which` failed     at getNotFoundError (C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\which\which.js:13:12)
gyp verb `which` failed     at F (C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\which\which.js:68:19)
gyp verb `which` failed     at E (C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\which\which.js:80:29)
gyp verb `which` failed     at C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\which\which.js:89:16
gyp verb `which` failed     at C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\isexe\index.js:42:5
gyp verb `which` failed     at C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\isexe\windows.js:36:5
gyp verb `which` failed     at FSReqCallback.oncomplete (fs.js:158:21) {
gyp verb `which` failed   stack: 'Error: not found: python\n' +
gyp verb `which` failed     '    at getNotFoundError (C:\\Users\\umar.safeer\\Desktop\\test\\ng-multiselect-dropdown-master\\node_modules\\which\\which.js:13:12)\n' +
gyp verb `which` failed     '    at F (C:\\Users\\umar.safeer\\Desktop\\test\\ng-multiselect-dropdown-master\\node_modules\\which\\which.js:68:19)\n' +
gyp verb `which` failed     '    at E (C:\\Users\\umar.safeer\\Desktop\\test\\ng-multiselect-dropdown-master\\node_modules\\which\\which.js:80:29)\n' +
gyp verb `which` failed     '    at C:\\Users\\umar.safeer\\Desktop\\test\\ng-multiselect-dropdown-master\\node_modules\\which\\which.js:89:16\n' +
gyp verb `which` failed     '    at C:\\Users\\umar.safeer\\Desktop\\test\\ng-multiselect-dropdown-master\\node_modules\\isexe\\index.js:42:5\n' +
gyp verb `which` failed     '    at C:\\Users\\umar.safeer\\Desktop\\test\\ng-multiselect-dropdown-master\\node_modules\\isexe\\windows.js:36:5\n' +
gyp verb `which` failed     '    at FSReqCallback.oncomplete (fs.js:158:21)',
gyp verb `which` failed   code: 'ENOENT'
gyp verb `which` failed }
gyp verb could not find "python". checking python launcher
gyp verb could not find "python". guessing location
gyp verb ensuring that file exists: C:\Python27\python.exe
gyp ERR! configure error
gyp ERR! stack Error: Can't find Python executable "python", you can set the PYTHON env variable.
gyp ERR! stack     at PythonFinder.failNoPython (C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\node-gyp\lib\configure.js:484:19)
gyp ERR! stack     at PythonFinder.<anonymous> (C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\node-gyp\lib\configure.js:509:16)
gyp ERR! stack     at callback (C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\graceful-fs\polyfills.js:295:20)
gyp ERR! stack     at FSReqCallback.oncomplete (fs.js:158:21)
gyp ERR! System Windows_NT 10.0.18362
gyp ERR! command "C:\\Program Files\\nodejs\\node.exe" "C:\\Users\\umar.safeer\\Desktop\\test\\ng-multiselect-dropdown-master\\node_modules\\node-gyp\\bin\\node-gyp.js" "rebuild" "--verbose" "--libsass_ext=" "--libsass_cflags=" "--libsass_ldflags=" "--libsass_library="
gyp ERR! cwd C:\Users\umar.safeer\Desktop\test\ng-multiselect-dropdown-master\node_modules\node-sass
gyp ERR! node -v v12.13.1
gyp ERR! node-gyp -v v3.8.0
gyp ERR! not ok
Build failed with error code: 1
npm WARN @angular/core@8.2.2 requires a peer of zone.js@~0.9.1 but none is installed. You must install peer dependencies yourself.
npm WARN @angular/http@7.2.15 requires a peer of @angular/core@7.2.15 but none is installed. You must install peer dependencies yourself.
npm WARN @angular/http@7.2.15 requires a peer of @angular/platform-browser@7.2.15 but none is installed. You must install peer dependencies yourself.
npm WARN @angular/tsc-wrapped@4.4.7 requires a peer of typescript@^2.1.5 but none is installed. You must install peer dependencies yourself.
npm WARN angular2-markdown@2.2.3 requires a peer of rxjs@^5.5.2 but none is installed. You must install peer dependencies yourself.
npm WARN angular2-markdown@2.2.3 requires a peer of @angular/core@^2.4.0 || ^4.0.3 || ^5.0.0 but none is installed. You must install peer dependencies yourself.
npm WARN axobject-query@2.1.1 requires a peer of eslint@^5 || ^6 but none is installed. You must install peer dependencies yourself.
npm WARN karma-jasmine-html-reporter@1.5.2 requires a peer of jasmine-core@>=3.5 but none is installed. You must install peer dependencies yourself.
npm WARN tsickle@0.21.6 requires a peer of typescript@^2.1.0 but none is installed. You must install peer dependencies yourself.
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.11 (node_modules\webpack-dev-server\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.11: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.11 (node_modules\watchpack\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.11: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.11 (node_modules\@angular\compiler-cli\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.11: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@1.2.11 (node_modules\jest-haste-map\node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@1.2.11: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.1.2 (node_modules\fsevents):
npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.1.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})

npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! node-sass@4.5.3 postinstall: `node scripts/build.js`
npm ERR! Exit status 1
npm ERR!
npm ERR! Failed at the node-sass@4.5.3 postinstall script.
npm ERR! This is probably not a problem with npm. There is likely additional logging output above.

npm ERR! A complete log of this run can be found in:
npm ERR!     C:\Users\umar.safeer\AppData\Roaming\npm-cache\_logs\2020-02-04T11_45_43_325Z-debug.log
```

Yeah I'm having similar problems.
The packages need to be cleaned up.

Seems there is a package angular-library-builder that is old and deprecated. (it's not used by build:lib script either), which installs a really old version of node sass as a dependency, which is meant for node 8.x, but other packages (like angular), require node 10.x+.

If I remove that package from package.json and install (with either node 10.x or 12.x) it gets passed the sass issues but then I get:
```
ERROR in app/app.module.ts:5:43 - error TS2307: Cannot find module 'ngx-bootstrap'.

5 import { TabsModule, ButtonsModule } from 'ngx-bootstrap';
                                            ~~~~~~~~~~~~~~~

ERROR in app/app.module.ts(23,5): Error during template compile of 'AppModule'
  Could not resolve ngx-bootstrap relative to [object Object]..

```

--- a/package.json
+++ b/package.json
@@ -66,7 +66,6 @@
     "@types/jasminewd2": "~2.0.2",
     "@types/node": "~12.7.2",
     "angular-cli-ghpages": "^0.6.0-rc.2",
-    "angular-library-builder": "^1.5.12",
     "angular2-markdown": "^2.2.3",
     "codelyzer": "~5.1.0",
     "copyfiles": "^2.0.0",

--- a/src/app/app.module.ts
+++ b/src/app/app.module.ts
@@ -2,7 +2,8 @@ import { NgModule } from '@angular/core';
 import { FormsModule, ReactiveFormsModule } from '@angular/forms';
 import { BrowserModule } from '@angular/platform-browser';
 
-import { TabsModule, ButtonsModule } from 'ngx-bootstrap';
+import { TabsModule } from 'ngx-bootstrap/tabs';
+import { ButtonsModule } from 'ngx-bootstrap/buttons';
 import { NgMultiSelectDropDownModule } from '../ng-multiselect-dropdown/src';
 // import { NgMultiSelectDropDownModule } from 'ng-multiselect-dropdown';
 

